src/components/numberInput/index.tsx

```diff
@@ -111,50 +143,54 @@ function NumberInput(props: NumberInputProps, ref: any) {
     }
   }, [hasText, trailingText, trailingTextStyle]);
 
-  const _containerStyle = useMemo(() => {
-    return [styles.containerStyle, containerStyle];
-  }, [containerStyle]);
-
-  const _onChangeText = useCallback((text: string) => {
+  const onChangeText = useCallback(async (text: string) => {
     processInput(text);
   },
   [processInput]);
 
-  const value = useMemo(() => {
-    return data?.type === 'valid' || data?.type === 'error' ? data.userInput : '';
-  }, [data]);
-
   const formatter = useCallback(() => {
     return data?.type === 'valid' ? data.formattedNumber : data?.type === 'error' ? data.userInput : '';
   }, [data]);
 
-  // Fixing RN bug in Android (placeholder + trailingText) - https://github.com/facebook/react-native/issues/35611
-  const _placeholder = useMemo(() => {
-    return isEmpty(value) ? placeholder : undefined;
-  }, [placeholder, value]);
+  const renderNumberInput = useCallback((value?: string) => {
+    return (
+      <View row style={containerStyle}>
+        {/* @ts-expect-error ref is somehow required, not sure from where */}
```

Comment:
Maybe we should open a ticket to check that? I guess our users will get it too in some cases.

Reply:
TBH, I doubt wi'll get to it unless it comes from a user

Reply:
Let's add it to the after-hours sync, I don't think it's a good practice to add those `@ts-expect-error`s without understanding it.

Reply:
This is outdated anyway